# to-do Auswertungen
- partei
- Zeiträume
- Mandat-Art
- Sektor

In [128]:
import pandas as pd
import plotly.graph_objects as go
import datetime as dt
from plotly.subplots import make_subplots

In [129]:
sidejobs = pd.read_excel("files\\sidejobs.xlsx")
try:
    sidejobs.drop(columns=["Unnamed: 0"], axis=1, inplace=True)
except:
    pass
#dt.datetime.fromtimestamp(1495534675)
sidejobs["created_at"] = sidejobs["created"].apply(lambda x: dt.datetime.fromtimestamp(x))
#sidejobs["created"].astype(str).astype('datetime64[ns]')

In [130]:
sidejobs.columns

Index(['id', 'entity_type', 'label', 'api_url', 'job_title_extra', 'category',
       'income_level', 'income', 'interval', 'data_change_date',
       'additional_information', 'created', 'field_city', 'income_total',
       'mandate_1', 'mandate_2', 'mandate_3', 'organization', 'topic_1',
       'topic_2', 'topic_3', 'topic_4', 'topic_5', 'topic_6', 'topic_7',
       'topic_8', 'topic_9', 'country', 'created_at'],
      dtype='object')

In [131]:
candidacies_mandates = pd.read_excel("files\\candidacies_mandates.xlsx")

#candidacies_mandates.drop(columns=["Unnamed: 0"], axis=1, inplace=True)

In [132]:
cand_mandates_relevant_cols = ['label', 'fraction_membership1', 'politician', 'electoral_list', 'constituency', 'parliament_period']

In [133]:
parliament_periods = pd.read_excel("files\\parliament_periods.xlsx", converters={'start_date_period':pd.to_datetime,'end_date_period':pd.to_datetime} )
parliament_periods.drop(columns=["Unnamed: 0"], axis=1, inplace=True)
parliament_periods.head()


id        entity_type                     label  \
0  158  parliament_period   Brandenburg 2024 - 2029   
1  157  parliament_period       Sachsen 2024 - 2029   
2  156  parliament_period     Thüringen 2024 - 2029   
3  155  parliament_period  EU-Parlament 2024 - 2029   
4  154  parliament_period     Brandenburg Wahl 2024   

                                             api_url  \
0  https://www.abgeordnetenwatch.de/api/v2/parlia...   
1  https://www.abgeordnetenwatch.de/api/v2/parlia...   
2  https://www.abgeordnetenwatch.de/api/v2/parlia...   
3  https://www.abgeordnetenwatch.de/api/v2/parlia...   
4  https://www.abgeordnetenwatch.de/api/v2/parlia...   

                               abgeordnetenwatch_url  \
0       https://www.abgeordnetenwatch.de/brandenburg   
1           https://www.abgeordnetenwatch.de/sachsen   
2        https://www.abgeordnetenwatch.de/thueringen   
3                https://www.abgeordnetenwatch.de/eu   
4  https://www.abgeordnetenwatch.de/brandenburg/w...   

                 parliament  \
0   Brandenburg 2024 - 2029   
1       Sachsen 2024 - 2029   
2     Thüringen 2024 - 2029   
3  EU-Parlament 2024 - 2029   
4     Brandenburg Wahl 2024   

                                     previous_period         type  \
0  {'id': 154, 'entity_type': 'parliament_period'...  legislature   
1  {'id': 153, 'entity_type': 'parliament_period'...  legislature   
2  {'id': 152, 'entity_type': 'parliament_period'...  legislature   
3  {'id': 151, 'entity_type': 'parliament_period'...  legislature   
4                                                NaN     election   

  election_date start_date_period end_date_period  
0           NaN        2024-10-17      2029-10-22  
1           NaN        2024-10-01      2029-09-30  
2           NaN        2024-09-26      2029-09-30  
3           NaN        2024-07-16      2029-07-15  
4    2024-09-22        2024-08-15      2024-09-30

In [134]:
parliament_periods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 140 entries, 0 to 139
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   id                     140 non-null    int64         
 1   entity_type            140 non-null    object        
 2   label                  140 non-null    object        
 3   api_url                140 non-null    object        
 4   abgeordnetenwatch_url  140 non-null    object        
 5   parliament             140 non-null    object        
 6   previous_period        71 non-null     object        
 7   type                   140 non-null    object        
 8   election_date          73 non-null     object        
 9   start_date_period      140 non-null    datetime64[ns]
 10  end_date_period        140 non-null    datetime64[ns]
dtypes: datetime64[ns](2), int64(1), object(8)
memory usage: 12.2+ KB


In [135]:
parliament_periods[["label", "parliament", "type", "start_date_period", "end_date_period"]]

label                   parliament         type  \
0        Brandenburg 2024 - 2029      Brandenburg 2024 - 2029  legislature   
1            Sachsen 2024 - 2029          Sachsen 2024 - 2029  legislature   
2          Thüringen 2024 - 2029        Thüringen 2024 - 2029  legislature   
3       EU-Parlament 2024 - 2029     EU-Parlament 2024 - 2029  legislature   
4          Brandenburg Wahl 2024        Brandenburg Wahl 2024     election   
..                           ...                          ...          ...   
135    Rheinland-Pfalz Wahl 2006    Rheinland-Pfalz Wahl 2006     election   
136  Baden-Württemberg Wahl 2006  Baden-Württemberg Wahl 2006     election   
137          Bundestag Wahl 2005          Bundestag Wahl 2005     election   
138             Berlin Wahl 2011             Berlin Wahl 2011     election   
139       EU-Parlament Wahl 2014       EU-Parlament Wahl 2014     election   

    start_date_period end_date_period  
0          2024-10-17      2029-10-22  
1          2024-10-01      2029-09-30  
2          2024-09-26      2029-09-30  
3          2024-07-16      2029-07-15  
4          2024-08-15      2024-09-30  
..                ...             ...  
135        2006-02-18      2006-03-25  
136        2006-02-25      2006-03-25  
137        2005-08-18      2005-09-18  
138        2011-08-17      2011-09-17  
139        2014-04-25      2014-05-25  

[140 rows x 5 columns]

In [136]:
parl_period_relevant_cols = ["label", "parliament", "type", "start_date_period", "end_date_period"]

In [137]:
mandates_parliaments_merge = candidacies_mandates[cand_mandates_relevant_cols].merge(parliament_periods[parl_period_relevant_cols], left_on="parliament_period", right_on="parliament", how="left")
mandates_parliaments_merge.head()

label_x  \
0      Dirk Beyer (Rheinland-Pfalz 2021 - 2026)   
1     Lucia Schanbacher (Bundestag 2021 - 2025)   
2  Anne Riecke (Schleswig-Holstein 2022 - 2027)   
3   Tamina Reinecke (Niedersachsen 2022 - 2027)   
4          Katja Maurer (Thüringen 2024 - 2029)   

                    fraction_membership1         politician  \
0                    SPD seit 01.01.2025         Dirk Beyer   
1                    SPD seit 01.01.2025  Lucia Schanbacher   
2                    FDP seit 13.12.2024        Anne Riecke   
3  Bündnis 90/Die Grünen seit 12.12.2024    Tamina Reinecke   
4                              DIE LINKE       Katja Maurer   

                                      electoral_list  \
0                                                NaN   
1  Landesliste Baden-Württemberg (Bundestag 2021 ...   
2   Landesliste FDP (Schleswig-Holstein 2022 - 2027)   
3  Landesliste Bündnis 90/Die Grünen (Niedersachs...   
4      Landesliste DIE LINKE (Thüringen 2024 - 2029)   

                                        constituency  \
0           33 - Worms (Rheinland-Pfalz 2021 - 2026)   
1          258 - Stuttgart I (Bundestag 2021 - 2025)   
2  6 - Dithmarschen-Schleswig (Schleswig-Holstein...   
3          8 - Helmstedt (Niedersachsen 2022 - 2027)   
4             25 - Erfurt II (Thüringen 2024 - 2029)   

                parliament_period                         label_y  \
0     Rheinland-Pfalz 2021 - 2026     Rheinland-Pfalz 2021 - 2026   
1           Bundestag 2021 - 2025           Bundestag 2021 - 2025   
2  Schleswig-Holstein 2022 - 2027  Schleswig-Holstein 2022 - 2027   
3       Niedersachsen 2022 - 2027       Niedersachsen 2022 - 2027   
4           Thüringen 2024 - 2029           Thüringen 2024 - 2029   

                       parliament         type start_date_period  \
0     Rheinland-Pfalz 2021 - 2026  legislature        2021-05-18   
1           Bundestag 2021 - 2025  legislature        2021-09-29   
2  Schleswig-Holstein 2022 - 2027  legislature        2022-05-16   
3       Niedersachsen 2022 - 2027  legislature        2022-11-08   
4           Thüringen 2024 - 2029  legislature        2024-09-26   

  end_date_period  
0      2026-03-13  
1      2025-10-26  
2      2027-05-30  
3      2027-11-18  
4      2029-09-30

In [138]:
mandates_parliaments = mandates_parliaments_merge[["label_x","fraction_membership1","politician", "constituency", "parliament_period", "type", "start_date_period", "end_date_period"]].copy()

In [139]:
sidejobs_pp_1 = sidejobs.merge(mandates_parliaments[["label_x", "start_date_period", "end_date_period"]], left_on=["mandate_1"], right_on=["label_x"], how="left", suffixes=(None, '_1'))
sidejobs_pp_2 = sidejobs_pp_1.merge(mandates_parliaments[["label_x", "start_date_period", "end_date_period"]], left_on=["mandate_2"], right_on=["label_x"], how="left", suffixes=(None, '_2'))
sidejobs_pp_3 = sidejobs_pp_2.merge(mandates_parliaments[["label_x", "start_date_period", "end_date_period"]], left_on=["mandate_3"], right_on=["label_x"], how="left", suffixes=(None, '_3'))


In [140]:
sidejobs_pp_3.columns

Index(['id', 'entity_type', 'label', 'api_url', 'job_title_extra', 'category',
       'income_level', 'income', 'interval', 'data_change_date',
       'additional_information', 'created', 'field_city', 'income_total',
       'mandate_1', 'mandate_2', 'mandate_3', 'organization', 'topic_1',
       'topic_2', 'topic_3', 'topic_4', 'topic_5', 'topic_6', 'topic_7',
       'topic_8', 'topic_9', 'country', 'created_at', 'label_x',
       'start_date_period', 'end_date_period', 'label_x_2',
       'start_date_period_2', 'end_date_period_2', 'label_x_3',
       'start_date_period_3', 'end_date_period_3'],
      dtype='object')

In [141]:
sidejobs_merge = sidejobs_pp_3.loc[:,~sidejobs_pp_3.columns.str.startswith('label')].copy()
sidejobs_merge.head()

id entity_type                                            api_url  \
0  49     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
1  48     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
2  47     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
3  46     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
4  45     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   

  job_title_extra  category  income_level  income  interval data_change_date  \
0             NaN     29229           NaN     NaN       NaN       2018-11-19   
1             NaN     29229           NaN     NaN       NaN       2018-11-19   
2             NaN     29230           NaN     NaN       NaN       2022-11-08   
3             NaN     29230           NaN     NaN       NaN       2024-09-23   
4             NaN     29230           NaN     NaN       NaN       2018-11-19   

  additional_information  ...  topic_8 topic_9      country  \
0                    NaN  ...      NaN     NaN  Deutschland   
1                    NaN  ...      NaN     NaN  Deutschland   
2                    NaN  ...      NaN     NaN  Deutschland   
3                    NaN  ...      NaN     NaN  Deutschland   
4                    NaN  ...      NaN     NaN  Deutschland   

           created_at start_date_period end_date_period start_date_period_2  \
0 2017-05-23 12:17:55               NaT             NaT                 NaT   
1 2017-05-23 12:16:37               NaT             NaT                 NaT   
2 2017-05-23 09:28:27               NaT             NaT                 NaT   
3 2017-05-23 09:27:58               NaT             NaT                 NaT   
4 2017-05-23 09:27:10               NaT             NaT                 NaT   

  end_date_period_2 start_date_period_3 end_date_period_3  
0               NaT                 NaT               NaT  
1               NaT                 NaT               NaT  
2               NaT          2021-09-29        2025-10-26  
3               NaT          2021-09-29        2025-10-26  
4               NaT                 NaT               NaT  

[5 rows x 34 columns]

In [142]:
sidejobs_merge[["created_at","start_date_period", "end_date_period","start_date_period_2"]].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16236 entries, 0 to 16235
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   created_at           16236 non-null  datetime64[ns]
 1   start_date_period    6133 non-null   datetime64[ns]
 2   end_date_period      6133 non-null   datetime64[ns]
 3   start_date_period_2  752 non-null    datetime64[ns]
dtypes: datetime64[ns](4)
memory usage: 507.5 KB


In [143]:
sidejobs_merge.columns

Index(['id', 'entity_type', 'api_url', 'job_title_extra', 'category',
       'income_level', 'income', 'interval', 'data_change_date',
       'additional_information', 'created', 'field_city', 'income_total',
       'mandate_1', 'mandate_2', 'mandate_3', 'organization', 'topic_1',
       'topic_2', 'topic_3', 'topic_4', 'topic_5', 'topic_6', 'topic_7',
       'topic_8', 'topic_9', 'country', 'created_at', 'start_date_period',
       'end_date_period', 'start_date_period_2', 'end_date_period_2',
       'start_date_period_3', 'end_date_period_3'],
      dtype='object')

In [144]:
sidejobs_merge.loc[0,"created_at"], sidejobs_merge.loc[0,"start_date_period"]

(Timestamp('2017-05-23 12:17:55'), NaT)

In [149]:
def get_relevant_period(row):
    print(row["created_at"])
    print(row["start_date_period"])
    
    if row["created_at"] >= row["start_date_period"] and row["created_at"] <= row["end_date_period"]:
        
        return row["mandate_1"]
    elif row["created_at"] >= row["start_date_period_2"] and row["created_at"] <= row["end_date_period_2"]:
        return row["mandate_2"]
    elif row["created_at"] >= row["start_date_period_3"] and row["created_at"] <= row["end_date_period_3"]:
        return row["mandate_3"]
    else:
        return None
    


In [150]:
sidejobs_merge['relevant_period'] = sidejobs_merge.apply(lambda x: get_relevant_period(x), axis=1)

2017-05-23 12:17:55
NaT
2017-05-23 12:16:37
NaT
2017-05-23 09:28:27
NaT
2017-05-23 09:27:58
NaT
2017-05-23 09:27:10
NaT
2017-05-23 09:26:52
NaT
2017-05-23 09:26:33
NaT
2017-05-23 09:26:14
NaT
2017-05-23 09:25:57
NaT
2017-05-23 09:25:37
NaT
2017-05-23 09:25:18
NaT
2017-05-23 09:24:57
NaT
2017-05-23 09:24:40
NaT
2017-05-23 09:24:23
NaT
2017-05-23 09:24:05
NaT
2017-05-23 09:23:40
NaT
2017-05-23 09:11:42
NaT
2017-05-23 09:11:05
NaT
2017-05-23 09:10:29
NaT
2017-05-23 09:09:55
NaT
2017-05-23 09:09:21
NaT
2017-05-23 09:08:47
NaT
2017-05-23 09:08:14
NaT
2017-05-23 09:07:41
NaT
2017-05-23 09:07:08
NaT
2017-05-23 09:06:12
NaT
2017-05-23 09:05:39
NaT
2017-05-23 09:04:57
NaT
2017-05-23 09:03:58
NaT
2017-05-23 09:03:22
NaT
2017-05-23 09:02:41
NaT
2017-05-23 09:02:00
NaT
2017-05-23 09:01:09
NaT
2017-05-23 08:55:14
NaT
2017-05-23 08:54:55
NaT
2017-05-23 08:54:12
NaT
2017-05-23 08:52:50
NaT
2017-05-23 08:50:19
NaT
2017-05-23 08:49:45
NaT
2017-05-23 08:49:23
NaT
2017-05-23 08:48:57
NaT
2017-05-23 08:48

In [156]:
sidejobs_merge.merge(mandates_parliaments, left_on=["relevant_period"], right_on=["label_x"], how="left", suffixes=(None, '_rel'))

id entity_type                                            api_url  \
0         49     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
1         48     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
2         47     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
3         46     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
4         45     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
...      ...         ...                                                ...   
16231  18454     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
16232  18453     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
16233  18452     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
16234  18451     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   
16235  18450     sidejob  https://www.abgeordnetenwatch.de/api/v2/sidejo...   

              job_title_extra  category  income_level   income  interval  \
0                         NaN     29229           NaN      NaN       NaN   
1                         NaN     29229           NaN      NaN       NaN   
2                         NaN     29230           NaN      NaN       NaN   
3                         NaN     29230           NaN      NaN       NaN   
4                         NaN     29230           NaN      NaN       NaN   
...                       ...       ...           ...      ...       ...   
16231                     NaN     29230           NaN      NaN       NaN   
16232  Einkommen im Jahr 2024     29232           2.0  4825.49       NaN   
16233  Einkommen im Jahr 2024     29228           1.0  1785.00       NaN   
16234  Einkommen im Jahr 2024     29232           1.0  3083.60       NaN   
16235                     NaN     29229           NaN      NaN       NaN   

      data_change_date additional_information  ...  end_date_period_3  \
0           2018-11-19                    NaN  ...                NaT   
1           2018-11-19                    NaN  ...                NaT   
2           2022-11-08                    NaN  ...         2025-10-26   
3           2024-09-23                    NaN  ...         2025-10-26   
4           2018-11-19                    NaN  ...                NaT   
...                ...                    ...  ...                ...   
16231       2024-12-31                    NaN  ...                NaT   
16232       2024-12-31                    NaN  ...                NaT   
16233       2024-12-31                 Brutto  ...                NaT   
16234       2024-12-31                    NaN  ...                NaT   
16235       2024-12-31                    NaN  ...                NaT   

                                     relevant_period  \
0                                               None   
1                                               None   
2                                               None   
3                                               None   
4                                               None   
...                                              ...   
16231            Serap Güler (Bundestag 2021 - 2025)   
16232  Michael Grosse-Brömer (Bundestag 2021 - 2025)   
16233        Michael Frieser (Bundestag 2021 - 2025)   
16234       Johannes Fechner (Bundestag 2021 - 2025)   
16235       Heike Engelhardt (Bundestag 2021 - 2025)   

                                             label_x fraction_membership1  \
0                                                NaN                  NaN   
1                                                NaN                  NaN   
2                                                NaN                  NaN   
3                                                NaN                  NaN   
4                                                NaN                  NaN   
...                                              ...                  ...   
16231            Serap Güler (Bundestag 2021 - 2025)       

In [22]:

sidejobs[["created_at","mandate_1","mandate_2", "mandate_3"]].head()

created_at                                  mandate_1  \
0 2017-05-23 12:17:55  Astrid Freudenstein (Bundestag 2013-2017)   
1 2017-05-23 12:16:37     Barbara Woltmann (Bundestag 2013-2017)   
2 2017-05-23 09:28:27      Reinhard Brandl (Bundestag 2013-2017)   
3 2017-05-23 09:27:58      Reinhard Brandl (Bundestag 2013-2017)   
4 2017-05-23 09:27:10         Gernot Erler (Bundestag 2013-2017)   

                                     mandate_2  \
0  Astrid Freudenstein (Bundestag 2017 - 2021)   
1                                          NaN   
2      Reinhard Brandl (Bundestag 2017 - 2021)   
3      Reinhard Brandl (Bundestag 2017 - 2021)   
4                                          NaN   

                                 mandate_3  
0                                      NaN  
1                                      NaN  
2  Reinhard Brandl (Bundestag 2021 - 2025)  
3  Reinhard Brandl (Bundestag 2021 - 2025)  
4                                      NaN

In [6]:
sidejobs.merge(candidacies_mandates[cand_mandates_relevant_cols], candidacies_mandates[cand_mandates_relevant_cols])

NameError: name 'cand_mandates_relevant_cols' is not defined

In [7]:
sidejobs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16236 entries, 0 to 16235
Data columns (total 28 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      16236 non-null  int64  
 1   entity_type             16236 non-null  object 
 2   label                   16236 non-null  object 
 3   api_url                 16236 non-null  object 
 4   job_title_extra         7470 non-null   object 
 5   category                16236 non-null  int64  
 6   income_level            7498 non-null   float64
 7   income                  3027 non-null   float64
 8   interval                702 non-null    float64
 9   data_change_date        15600 non-null  object 
 10  additional_information  2987 non-null   object 
 11  created                 16236 non-null  int64  
 12  field_city              6920 non-null   object 
 13  income_total            2880 non-null   object 
 14  mandate_1               16236 non-null

In [8]:
sidejobs[["mandate_1", "mandate_2", "mandate_3", "created", "data_change_date"]].head()

mandate_1  \
0  Astrid Freudenstein (Bundestag 2013-2017)   
1     Barbara Woltmann (Bundestag 2013-2017)   
2      Reinhard Brandl (Bundestag 2013-2017)   
3      Reinhard Brandl (Bundestag 2013-2017)   
4         Gernot Erler (Bundestag 2013-2017)   

                                     mandate_2  \
0  Astrid Freudenstein (Bundestag 2017 - 2021)   
1                                          NaN   
2      Reinhard Brandl (Bundestag 2017 - 2021)   
3      Reinhard Brandl (Bundestag 2017 - 2021)   
4                                          NaN   

                                 mandate_3     created data_change_date  
0                                      NaN  1495534675       2018-11-19  
1                                      NaN  1495534597       2018-11-19  
2  Reinhard Brandl (Bundestag 2021 - 2025)  1495524507       2022-11-08  
3  Reinhard Brandl (Bundestag 2021 - 2025)  1495524478       2024-09-23  
4                                      NaN  1495524430       2018-11-19

In [43]:
sidejobs["created"]

0        1495534675
1        1495534597
2        1495524507
3        1495524478
4        1495524430
            ...    
16231    1735658440
16232    1735658209
16233    1735657704
16234    1735657491
16235    1735656992
Name: created, Length: 16236, dtype: int64

In [5]:
sidejobs["entity_type"].value_counts()

entity_type
sidejob    16236
Name: count, dtype: int64

In [11]:
pd.DataFrame(sidejobs["label"].value_counts()).reset_index().sort_values(by="count", ascending=False)

label  count
0                              Mitglied des Kuratoriums    380
1                           Mitglied des Aufsichtsrates    364
2                                 Mitglied des Beirates    317
3                                           Beteiligung    310
4                               Mitglied des Vorstandes    239
...                                                 ...    ...
7649               Stellv. Mitglied der Umlandkonferenz      1
7650  Mitglied des Beirates für Schulentwicklungspla...      1
7651  Stellv. Mitglied des Betriebsausschusses (bis ...      1
7652  Mitglied des Vorstandes des Eisenbahninfrastru...      1
7653              Mitglied des Beirates (ab 09.11.2021)      1

[7654 rows x 2 columns]

Kategorie
- 29231 = Beteiligung an Kapital- oder Personengesellschaften
- 29647 = Entgeltliche Tätigkeiten neben dem Mandat
- 29229 = Funktionen in Körperschaften und Anstalten des öffentlichen Rechts
- 29228 = Funktionen in Unternehmen
- 29230 = Funktionen in Vereinen, Verbänden und Stiftungen
- 29232 = Spenden/Zuwendungen für politische Tätigkeit
- 29233 = Vereinbarungen über künftige Tätigkeiten oder Vermögensvorteile
- 29234 = Berufliche Tätigkeit vor der Mitgliedschaft im Deutschen Bundestag


In [12]:
pd.DataFrame(sidejobs["category"].value_counts()).reset_index().sort_values(by="count", ascending=False)

category  count
0     29647   6967
1     29230   3816
2     29229   2731
3     29228   1706
4     29234    475
5     29231    371
6     29232    163
7     29233      7

Einkommenslevel
- 0 = 1 € bis 1.000 €
- 1 = 1.000 € bis 3.500 €
- 2 = 3.500 € bis 7.000 €
- 3 = 7.000 € bis 15.000 €
- 4 = 15.000 € bis 30.000 €
- 5 = 30.000 € bis 50.000 €
- 6 = 50.000 € bis 75.000 €
- 7 = 75.000 € bis 100.000 €
- 8 = 100.000 € bis 150.000 €
- 9 = 150.000 € bis 250.000 €
- 10 = ab 250.000 €


In [16]:
sidejobs["income_level"].value_counts().reset_index().sort_values(by="count", ascending=False)

income_level  count
0           1.0   3605
1           2.0   1551
2           3.0   1279
3           4.0    525
4           5.0    209
5           6.0     96
6          10.0     76
7           7.0     56
8           9.0     53
9           8.0     48

- Interval
    0 = einmalig
    1 = monatlich
    2 = jährlich

In [17]:
sidejobs["interval"].value_counts().reset_index().sort_values(by="count", ascending=False)


interval  count
0       1.0    555
1       2.0    147

In [22]:
fig = go.Figure(go.Histogram(
        x= sidejobs["income"],
        
))
fig.show()

In [24]:
sidejobs["income"].value_counts().reset_index().sort_values(by="income", ascending=False)                   


income  count
603   3928061.19      1
2300  3340173.16      1
2286  1380810.82      1
1163  1136160.72      1
589    998113.98      1
...          ...    ...
425        74.29      1
167        70.00      2
1871       64.00      1
56         60.00      4
205        50.00      1

[2385 rows x 2 columns]

In [15]:
sidejobs["label"].value_counts().reset_index().sort_values(by="count", ascending=False)                   

label  count
0                              Mitglied des Kuratoriums    380
1                           Mitglied des Aufsichtsrates    364
2                                 Mitglied des Beirates    317
3                                           Beteiligung    310
4                               Mitglied des Vorstandes    239
...                                                 ...    ...
7649               Stellv. Mitglied der Umlandkonferenz      1
7650  Mitglied des Beirates für Schulentwicklungspla...      1
7651  Stellv. Mitglied des Betriebsausschusses (bis ...      1
7652  Mitglied des Vorstandes des Eisenbahninfrastru...      1
7653              Mitglied des Beirates (ab 09.11.2021)      1

[7654 rows x 2 columns]

In [25]:
sidejobs.head()

id entity_type                                              label  \
0  49     sidejob                            Mitglied des Stadtrates   
1  48     sidejob                                   Ministerialrätin   
2  47     sidejob                        Vizepräsident, ehrenamtlich   
3  46     sidejob  Mitglied des Parlamentarischen Beirates, ehren...   
4  45     sidejob                                       Vorsitzender   

                                             api_url job_title_extra  \
0  https://www.abgeordnetenwatch.de/api/v2/sidejo...             NaN   
1  https://www.abgeordnetenwatch.de/api/v2/sidejo...             NaN   
2  https://www.abgeordnetenwatch.de/api/v2/sidejo...             NaN   
3  https://www.abgeordnetenwatch.de/api/v2/sidejo...             NaN   
4  https://www.abgeordnetenwatch.de/api/v2/sidejo...             NaN   

   category  income_level  income  interval data_change_date  ...  \
0     29229           NaN     NaN       NaN       2018-11-19  ...   
1     29229           NaN     NaN       NaN       2018-11-19  ...   
2     29230           NaN     NaN       NaN       2022-11-08  ...   
3     29230           NaN     NaN       NaN       2024-09-23  ...   
4     29230           NaN     NaN       NaN       2018-11-19  ...   

                 topic_1  topic_2 topic_3 topic_4 topic_5 topic_6 topic_7  \
0   Staat und Verwaltung      NaN     NaN     NaN     NaN     NaN     NaN   
1   Staat und Verwaltung      NaN     NaN     NaN     NaN     NaN     NaN   
2           Verteidigung      NaN     NaN     NaN     NaN     NaN     NaN   
3                Verkehr      NaN     NaN     NaN     NaN     NaN     NaN   
4  Bildung und Erziehung   Kultur     NaN     NaN     NaN     NaN     NaN   

  topic_8 topic_9      country  
0     NaN     NaN  Deutschland  
1     NaN     NaN  Deutschland  
2     NaN     NaN  Deutschland  
3     NaN     NaN  Deutschland  
4     NaN     NaN  Deutschland  

[5 rows x 28 columns]

In [7]:
fig = make_subplots(rows=1, cols=1)
fig.add_trace(
    go.Bar(x=pd.DataFrame(sidejobs["label"].value_counts()).reset_index()["label"], y=pd.DataFrame(sidejobs["label"].value_counts()).reset_index()["count"]))
